### PR TITLE
Basic bulk rename

### PR DESCRIPTION
**Pull request by tsujan**
Closes https://github.com/lxde/pcmanfm-qt/issues/599.

Follows and depends on https://github.com/lxde/libfm-qt/pull/136.

"#" is used for inserting the ascending numbers at any positions(s) of the base name but if omitted, the numbers will still be added appropriately.

The code could be extended later if needed.

### CODE REVIEW

File: pcmanfm/bulkrename.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            end = baseName.size();
+        }
+        baseName.insert(end, QLatin1Char('#'));
+    }
+    uint failed = 0;
+    for(auto& file: files) {
+        auto fileName = QString::fromStdString(file->name());
+        QString newName = baseName;
+        newName.replace(QLatin1Char('#'), QString::number(start));
+        if (newName == fileName || !Fm::changeFileName(file->path(), newName, nullptr, false)) {
```
Comment (palinek) [inline pcmanfm/bulkrename.cpp]: `Fm::changeFileName()` is blocking operation, so calling it in main thread can make the whole application "hang" (you also can get unresponsive desktop while long running renaming). This normally will not be a problem for (few) local files, but can be problematic with remote...

Comment (tsujan) [reply pcmanfm/bulkrename.cpp]: Thanks for mentioning that! I'll investigate it tomorrow -- today, I don't have time. However, for local files, I tested with > 1000 files and the renaming was instant; that was the reason I didn't include a progressbar in this.

Comment (tsujan) [reply pcmanfm/bulkrename.cpp]: In the meanwhile, feel free to add any commit.

**Comment by tsujan** [comment]
Added a progress dialog, which can be used for aborting bulk renaming (single renaming can't be canceled in libfm-qt yet).

File: pcmanfm/bulkrename.cpp [hunk 2]
```diff
@@ -68,17 +69,26 @@ BulkRenamer::BulkRenamer(const Fm::FileInfoList& files, QWidget* parent) {
...
-        newName.replace(QLatin1Char('#'), QString::number(start));
+        newName.replace(QLatin1Char('#'), QString::number(start + i));
         if (newName == fileName || !Fm::changeFileName(file->path(), newName, nullptr, false)) {
             ++failed;
         }
-        ++start;
+        ++i;
     }
-    if(failed == files.size()) {
+    progress.setValue(i);
```
Comment (palinek) [inline pcmanfm/bulkrename.cpp]: Shouldn't this be in the above loop?

Comment (tsujan) [reply pcmanfm/bulkrename.cpp]: No it shouldn't.

Comment (MggMuggins) [reply pcmanfm/bulkrename.cpp]: Line 76 has the same call.

Comment (tsujan) [reply pcmanfm/bulkrename.cpp]: > Line 76 has the same call.

Not the same. Line 90 is after the loop, when `i` is `files.size()`, which mean 100%.

Comment (MggMuggins) [reply pcmanfm/bulkrename.cpp]: I thought he was asking why a call that should be made every time the loop executes is being made at the end. It makes sense why you did it.

**Comment by agaida** [comment]
GTM

**Pull request merged**